### PR TITLE
Adds flag for users who haven't made contact in 30 days

### LINKED ISSUE #186: Flag volunteers who haven't had a successful case contact in 30 days
**What type of user is this for? volunteer/supervisor/admin/all**
Admin/supervisor viewing volunteers

**Description**
When a **supervisor or admin** views a list of **volunteers**, there should be a little 🕒(or similar) next to the volunteer name if they have not made any successful `case_contact` in the last 30 days

If the volunteer has multiple cases, show the icon if any one of their cases has no successful `case_contact`s - so a volunteer doing well on case1 and **not** on case2 **will** get the flag

Note: someday we will want to do this per *type* of case contact, but that is not part of this issue. For example, the volunteer should talk to the youth every other week at a minimum. But the volunteer does not need to talk to the therapist even every month. Maybe someday we will want this to be fully configurable

**Screenshots of current behavior, if any**
Screen Shot 2020-09-09 at 6 48 51 PM
Screen Shot 2020-09-09 at 6 48 26 PM

**Related issues, if any**

### PR DESCRIPTION
**Pull request by jmkoni**
### What github issue is this PR for, if any?
Resolves #186

### What changed, and why?
- paired with @keithrbennett
- added 🕐 (clock icon) if volunteer has any one case where they haven't made contact in 30 days

### How will this affect user permissions?
No permissions changed

### How is this tested? (please write tests!) 💖💪
Added tests on the model for each branch within the function

### Screenshots please :)
Screen Shot 2020-09-13 at 11 54 57 AM

### CODE REVIEW

**Review by compwron — APPROVED** [review]
Very very nice! ✨🕐✨

File: app/models/volunteer.rb [hunk 1]
```diff
@@ -37,6 +37,22 @@ def case_assignments_with_cases
...
+    current_contact_cases_count = cases_where_contact_made_in_days(num_days).count
+    return current_contact_cases_count == total_cases_count
+  end
+
+  private
+  def cases_where_contact_made_in_days(num_days = 30)
+    casa_cases
+      .joins(:case_contacts)
+      .where(case_contacts: { contact_made: true })
+      .where("case_contacts.occurred_at > ?", Date.today - num_days.days)
```
Comment (littleforest) [inline app/models/volunteer.rb:54]: I use `Date.current` per https://thoughtbot.com/blog/its-about-time-zones#a-summary-of-do39s-and-don39ts-with-time-zones

**Review by littleforest — CHANGES_REQUESTED** [review]
@jmkoni only one minor comment re which `Date` method to use!

**Review by littleforest — APPROVED** [review]
🎉

**Pull request merged**